### ISSUE TITLE
WCAG2ICT FPWD Comment from IBM: Inconsistency between 1.4.12 Text Spacing and 4.1.3 Status Messages

### ISSUE THREAD

**Issue by maryjom** [w3c/wcag2ict#216]
In WCAG2ICT both 1.4.12 Text Spacing and 4.1.3 Status Messages have the language where they only apply to "content implemented using markup language", but the guidance applies them differently to non-web software. These need to be made consistent, and it makes sense to apply the 1.4.12 Text Spacing requirement whenever software supports these text modifications.

4.1.3 allows this SC to be applicable to software - even when not implemented using markup language when it supports programmatic identification of status messages. Language is:
>In **[content implemented using markup languages, or that supports status message notifications]**, [status messages](https://www.w3.org/TR/WCAG22/#dfn-status-messages) can be [programmatically determined](https://w3c.github.io/wcag2ict/#dfn-programmatically-determined) through [role](https://w3c.github.io/wcag2ict/#dfn-role) or properties such that they can be presented to the user by [assistive technologies](https://w3c.github.io/wcag2ict/#dfn-assistive-technology) without receiving focus.

**1.4.12 Text Spacing** kept the limitation "in content implemented using markup language" but does not apply the SC more broadly to software that supports changing any of the text spacing options. Language is:
>**[For non-web documents or software]** content implemented using markup languages **[in a way that supports modification of]** the following [text](https://www.w3.org/TR/WCAG22/#dfn-text) [style properties](https://www.w3.org/TR/WCAG22/#dfn-style-properties), no loss of content or functionality occurs by setting all of the following and by changing no other style property:

**Suggested modification of 1.4.12 language:** 
I have a couple of potential options for handling this.

**Option 1**
>**[For non-web documents or software]** content **[implemented using markup languages or software in a way that supports modification of]** the following [text](https://www.w3.org/TR/WCAG22/#dfn-text) [style properties](https://www.w3.org/TR/WCAG22/#dfn-style-properties), no loss of content or functionality occurs by setting all of the following and by changing no other style property:

**Option 2**
>**[For non-web documents or software]** content **[implemented using markup languages, or that supports modification of]** the following [text](https://www.w3.org/TR/WCAG22/#dfn-text) [style properties](https://www.w3.org/TR/WCAG22/#dfn-style-properties), no loss of content or functionality occurs by setting all of the following and by changing no other style property:

**Comment by pday1** [comment]
I prefer the latter option as it slightly broader, but am happy with either possibilities

**Comment by maryjom** [comment]
I think these two suggestions can go in front of the TF for review. Will add to the survey due 31 January.

**Comment by bruce-usab** [comment]
If we get consensus on an approach for one, I agree that should be used for the other.  With 4.1.3, think I have sorted out the two oppositional approaches as [taken literally](https://docs.google.com/document/d/1ejuKX3cq06b3MY48h2wJPHJXfFYJLIjI0rUNf3PfoJQ/edit#heading=h.h23pqwfqw5cj) and [go big](https://docs.google.com/document/d/1ejuKX3cq06b3MY48h2wJPHJXfFYJLIjI0rUNf3PfoJQ/edit#heading=h.z2ugc5o5w7os).

If the TF can rule one or the other out, then we iterate.

**Comment by maryjom** [comment]
I've created a [Google doc for 1.4.12 Text Spacing](https://docs.google.com/document/d/1_tw8vvXEESl2ybOCWw59psPq7so1oNj4LSghU0hgCZM/edit?usp=sharing) so we can tweak the guidance for the consistent approach we developed during the 16 Feb. sub-group meeting. See the [minutes for 16 Feb.](https://www.w3.org/2024/02/16-wcag2ict-minutes.html)

**Comment by maryjom** [comment]
Here are the [survey results](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-Jan-public-responses/results#xq6) for the proposals in the survey description.

**Comment by bruce-usab** [comment]
If the survey were still open, what I would say is that [SC 1.4.12](https://www.w3.org/TR/WCAG22/#text-spacing) **can be used as-is _without any_ word substitution** whatsoever.  We can add notes for products with closed functionality if we like.
> In content implemented using markup languages that support the following text style properties...

SC 1.4.12 is already tightly scoped.  It works just fine _as-is_ for non-web documents and non-web software.

WCAG2ICT _might_ need to caveat the definition for [style properties](https://www.w3.org/TR/WCAG22/#dfn-style-properties).  Have we started to revisit updating [Comments on Definitions in WCAG 2 Glossary](https://www.w3.org/WAI/GL/wcag2ict/#wcag2ict_comments_definitions)?

**Comment by maryjom** [comment]
The TF had 2 different surveys regarding changes to the editor's draft with follow-up TF discussions that reached consensus on the text:
- [Survey on 1.4.12 Text spacing](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-text-spacing-and-images/results#xq1) - consensus reached, see the [minutes for the 29 Feb. meeting](https://www.w3.org/2024/02/29-wcag2ict-minutes#t05)
- [Survey on 4.1.3 Status Messages](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-Status-Messages-update/results) - consensus reached, see the [minutes for the 22 Feb. meeting](https://www.w3.org/2024/02/22-wcag2ict-minutes#t02)

General guidance and the content in the section on SC Problematic for Closed Functionality were updated.

The agreed changes were implemented in PRs #316 and #314.

**Comment by maryjom** [comment]
DRAFT WCAG2ICT answer:

Thank you for your review and comment on WCAG2ICT. The task force reviewed the language for the 1.4.12 Text Spacing and 4.1.3 Status Messages Success Criteria and agree that these should be changed to better align in their interpretation. We considered your proposed text, but upon further consideration decided to take an approach that better aligns with the [WCAG2ICT Task Force work statement's scope of work](https://www.w3.org/WAI/GL/task-forces/wcag2ict/work-statement#scope). In that document, it is considered "out of scope" to propose changes to WCAG 2.x or develop any normative requirements. Therefore, we determined the WCAG2ICT Task Force is not at liberty to expand the scope of the WCAG success criteria beyond what WCAG states - to "content implemented using markup languages". 

However, we do agree that it makes sense to more broadly apply to non-web software, and so new notes were developed to indicate that. This will inform outside standards development organizations that they should consider broadening the scope or writing an additional requirement that parallels the WCAG criterion, as they see fit.

You can find the changes to 1.4.12 Text Spacing in PR #316 and to 4.1.3 Status Messages in PR #314. Both have been merged into the editor's draft.

**Comment by bruce-usab** [comment]
In regards to a question to me from @maryjom [on the list](https://lists.w3.org/Archives/Public/public-wcag2ict-tf/2024Feb/0042.html)...

> WCAG2ICT _might_ need to caveat the definition for [style properties](https://www.w3.org/TR/WCAG22/#dfn-style-properties). Have we started to revisit updating [Comments on Definitions in WCAG 2 Glossary](https://www.w3.org/WAI/GL/wcag2ict/#wcag2ict_comments_definitions)?

It's fine.

**Comment by maryjom** [comment]
See [survey results](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-public-responses-3/results#xq1) indicating unanimous approval of the  [draft answer](https://github.com/w3c/wcag2ict/issues/216#issuecomment-1971842036) to this issue.

**Comment by mitchellevan** [comment]
To call attention to this message https://github.com/w3c/wcag2ict/issues/216#issuecomment-1971842036, I've filed the following issue against EN 301 549: [Apply SC 4.1.3 Status Messages to native code in software \#448](https://labs.etsi.org/rep/HF/en301549/-/issues/448)
